src/hotspot/share/gc/shenandoah/shenandoahHeapRegionCounters.cpp

```diff
@@ -68,19 +71,25 @@ ShenandoahHeapRegionCounters::ShenandoahHeapRegionCounters() :
       _regions_data[i] = PerfDataManager::create_long_variable(SUN_GC, data_name,
                                                                PerfData::U_None, CHECK);
     }
+
+    if (ShenandoahLogRegionSampling) {
+      _log_file = new ShenandoahLogFileOutput(ShenandoahRegionSamplingFile, _timestamp->get_value());
+      _log_file->initialize(tty);
+    }
   }
 }
 
 ShenandoahHeapRegionCounters::~ShenandoahHeapRegionCounters() {
   if (_name_space != NULL) FREE_C_HEAP_ARRAY(char, _name_space);
+  if (_log_file != NULL) FREE_C_HEAP_OBJ(_log_file);
 }
 
 void ShenandoahHeapRegionCounters::update() {
   if (ShenandoahRegionSampling) {
     jlong current = nanos_to_millis(os::javaTimeNanos());
     jlong last = _last_sample_millis;
     if (current - last > ShenandoahRegionSamplingRate &&
-            Atomic::cmpxchg(&_last_sample_millis, last, current) == last) {
+        Atomic::cmpxchg(&_last_sample_millis, last, current) == last) {
```

Comment:
Neither the previous nor the new version seems to align correctly.

Reply:
Fixed